server/interpreter.js

```diff
@@ -2880,16 +2650,7 @@ Interpreter.prototype['stepForInStatement'] = function() {
   // Third, find the property name for this iteration.
   if (state.name_ === undefined) {
     done: do {
-      if (state.object_.isPrimitive) {
-        for (var prop in state.object_.data) {
-          if (state.visited_.has(prop)) {
-            state.visited_.add(prop);
-            state.name_ = prop;
-            state.visited_.add(prop);
-            break done;
-          }
-        }
-      } else {
+      if (state.object_.isObject) {
         for (var prop in state.object_.properties) {
           if (state.visited_.has(prop)) {
```

Comment:
This test is inverted, which explains why all the for..in tests are failing at the moment.